browse/controllers/year.py

```diff
@@ -0,0 +1,103 @@
+"""Handle requests for info about one year of archive activity"""
+
+from datetime import date
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from werkzeug.exceptions import BadRequest
+from flask import current_app, url_for
+
+from arxiv import status, taxonomy
+from browse.domain.listing import MonthCount, ListingCountResponse
+from browse.controllers.list_page import get_listing_service
+from browse.controllers.years_operating import years_operating, stats_by_year
+
+
+def year_page(archive_id: str, year: int) -> Any:
+    """Get year page for archive.
+
+    Parameters
+    ----------
+    archive : str
+    Must be an arXiv archive identifier.
+
+    year: int
+    Must be a two or four digit year.
+
+    Returns
+    -------
+    dict
+        Search result response data.
+    int
+        HTTP status code.
+    dict
+        Headers to add to the response.
+    """
+    thisYear = date.today().year
+
+    if year is None:
+        year = thisYear
+        
+    if year > thisYear:
+        # 307 because year might be valid in the future
+        return {}, status.HTTP_307_TEMPORARY_REDIRECT, {'Location': '/'}
+
+    if year < 100:
+        if year >= 91:
+            year = 1900 + year
+        else:
+            year = 2000 + year
+
+    if archive_id not in taxonomy.ARCHIVES:
+        raise BadRequest("Unknown archvie.")
+    else:
+        archive = taxonomy.ARCHIVES[archive_id]
+
+    listing_service = get_listing_service(current_app)
+    month_listing = listing_service.monthly_counts(archive_id, year)
+
+    for month in month_listing['month_counts']:
+        month['art'] = ascii_art_month(archive_id, month)
+        yymm = month['year'] + month['month']
```

Comment:
Looks like integers are getting added here. Maybe:

```yymm: str = f"{y - 1900 if y < 2000 else y - 2000:02}{m:02}"``

Reply:
This is so funny. :+1: